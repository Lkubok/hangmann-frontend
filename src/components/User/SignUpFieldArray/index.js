import SignUpFieldArray from "./SignUpFieldArray";
export default SignUpFieldArray;
