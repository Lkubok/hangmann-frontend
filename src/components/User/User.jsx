import React, { Component } from "react";
import { connect } from "react-redux";
import SignForm from "./SignForm";
import history from "../../history";

import "./User.scss";

export class User extends Component {
  render() {
    const { isLogged } = this.props;
    return <>{isLogged ? history.push("/") : <SignForm />}</>;
  }
}

const mapStateToProps = state => ({
  isLogged: state.appParamsReducer.isLogged
});

const mapDispatchToProps = {};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(User);
