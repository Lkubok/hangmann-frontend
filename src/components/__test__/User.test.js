import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { User } from "../User/User";
Enzyme.configure({ adapter: new Adapter() });

describe("User component", () => {
  beforeEach(() => {});
  it("Should match snapshot", () => {
    const wrapper = shallow(<User />);
    expect(wrapper).toMatchSnapshot();
  });
});
