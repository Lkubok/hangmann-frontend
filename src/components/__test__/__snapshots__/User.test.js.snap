// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component Should match snapshot 1`] = `
<Fragment>
  <withRouter(Connect(WithFormik(SignForm))) />
</Fragment>
`;
